plugin/core/windows.py

```diff
@@ -350,11 +350,13 @@ def window(self) -> sublime.Window:
 
     def sessions(self, view: sublime.View, capability: Optional[str] = None) -> Generator[Session, None, None]:
         file_name = view.file_name() or ''
+        scope = view2scope(view)
         for sessions in self._sessions.values():
             for session in sessions:
-                if capability is None or capability in session.capabilities:
-                    if session.state == ClientStates.READY and session.handles_path(file_name):
-                        yield session
+                if session.config.match_document(scope):
```

Comment:
Style nit: Could filter the list in the `for` loop already? I don't like a lot of nesting. :)

Reply:
I've put all of the checks in `can_handle` of `Session`. Hope that makes it a bit better. I don't personally mind nested ifs.